seismicpro/tests/test_gather.py

```diff
@@ -197,6 +208,16 @@ def test_gather_copy(gather, ignore):
     compare_gathers(copy_gather, gather, check_types=True)
 
 
+@pytest.mark.parametrize('columns', ['offset', 'col_1', ['col_1'], ['col_1', 'col_2']])
+def test_gather_store_headers_to_survey(gather_with_cols, columns):
+    """test_gather_store_headers_to_survey"""
+    gather_with_cols.store_headers_to_survey(columns)
+    survey = gather_with_cols.survey
+    headers_from_survey = survey.headers.loc[gather_with_cols.index].sort_values(by="offset")
+    headers_from_gather = gather_with_cols.headers.sort_values(by="offset")
```

Comment:
With this sort on the `DataFrame` level you actually guarantee that equality bellow holds, even if you mess with trace positions in your action. 

```suggestion
    headers_from_survey = survey.headers.loc[gather_with_cols.index]
    headers_from_gather = gather_with_cols.headers
```